### ISSUE TITLE
[BUG]: Elements controls with floating labels are rendered incorrectly

### ISSUE THREAD

**Issue by metalmarker** [stripe/react-stripe-js#470]
### What happened?

I'm using `<Elements>` in my React app and interested in using floating labels . The problem is that the controls are rendered incorrectly, as can be seen below.

I'm passing the following `appearance` object inside the `options` object passed  to `<Elements>`:
```
 appearance = {
   theme: 'stripe',
   labels: 'floating'
 }
```

The result is this:
<img width="584" alt="Screenshot 2024-02-06 at 10 29 38 AM" src="https://github.com/stripe/react-stripe-js/assets/106110899/78502159-5bc9-472c-81cb-99b7bab3d969">

When I look on the CSS associated with the <input> of each field, it looks like there's a calculation errors in the
`--c-labelFloatingTransform`, `--c-labelRestingTransform` and `--c-inputFloatingPaddingTop` variables as they include a NaN. Once I manually override the NaNs variables, the controls render correctly.

<img width="386" alt="Screenshot 2024-02-06 at 10 29 01 AM" src="https://github.com/stripe/react-stripe-js/assets/106110899/46a57749-ef76-4602-9edf-0f361693a36b">  
  
_  

**Additional information:** 
- Removing `labels: floating` (i.e., labels above the controls) results in correctly rendered controls
- I'm using @stripe/stripe-js and stripe/react-stripe-js 2.4.0.

**Comment by brendanm-stripe** [comment]
Hey there, thanks for the report on this. We've been investigating. Can you confirm that your integration has the payment element inside a hidden element that you then make visible?

Assuming thats the case, we're working on a general fix but a workaround you might be able to use in the meantime is forcing an Appearance API recalculation after the container is visible using `elements.update({ appearance: currentAppearance })`.

**Comment by metalmarker** [comment]
> Can you confirm that your integration has the payment element inside a hidden element that you then make visible?

Confirmed. That was one fast root cause analysis! I'm hiding the contents of the entire page (which includes the PaymentElement) until after it sends the `onReady` event.

I'll try out the suggested workaround. Thank you.

**Comment by metalmarker** [comment]
@brendanm-stripe it looks this was resolved (I'm now on stripe-js@3.0.5 and react-stripe-js@2.4.0). 
Great job!

**Comment by tylersmith-stripe** [comment]
Yes, to follow-up here, we've fixed this issue for the Payment Element!

We're continuing to apply this fix for [Link Authentication Element](https://docs.stripe.com/js/element/link_authentication_element) and [Address Element](https://docs.stripe.com/js/element/address_element) as a fast-follow.

We'll update and close out this ticket when those are fixed as well.
